config/routes.rb

```diff
@@ -18,3 +20,15 @@
     end
   end
 end
+namespace :api do
+    namespace :v1 do
+      resources :users do
+        resources :posts do
+          collection do
+            get 'all'
+          end
+          resources :comments, only: [:index, :create]
+        end
+      end
+    end
```

Comment:
- [ ] There is an issue with the nesting of your namespaced API routes—they are currently outside of the Rails.application.routes.draw block, which is likely causing the error am encountering. Here's how you can modify your routes.rb file to fix the issue:
```
Rails.application.routes.draw do
  devise_for :users, controllers: { registrations: 'users/registrations' }
  
  # ... other routes ...

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create, :destroy] do
      resources :comments, only: [:new, :create, :destroy], controller: 'comments'
      resources :likes, only: [:create, :destroy], controller: 'likes'
    end
  end

  # Here, nest your API routes within the main block
  namespace :api do
    namespace :v1 do
      resources :users do
        resources :posts do
          collection do
            get 'all'
          end
          resources :comments, only: [:index, :create]
        end
      end
    end
  end
end
```

By moving the namespace :api block inside the Rails.application.routes.draw block, you ensure that the Rails routing DSL methods (like resources) are recognized and correctly processed. This should resolve the NoMethodError you were facing.